src/literal.h

```diff
@@ -444,7 +442,20 @@ class Literal {
   Literal avgrUInt(const Literal& other) const;
 };
 
-using Literals = SmallVector<Literal, 1>;
+struct Literals : SmallVector<Literal, 1> {
+  Literals() = default;
+  Literals(std::initializer_list<Literal> init)
+    : SmallVector<Literal, 1>(init){};
+  Type getType() {
+    std::vector<Type> types;
+    for (auto& val : *this) {
+      types.push_back(val.type);
+    }
+    return Type(types);
+  }
+  bool isNone() { return size() == 0; }
+  bool isConcrete() { return size() != 0; }
```

Comment:
What about a vector of `(unreachable, i32, f64)`?

If that should be impossible, how about asserting on that here?

Reply:
I'll add asserts in the constructors for `Literal` and `Literals`.